freedomofpress/securedrop-workstation
# Don't enable sdw-notify.timer unit via Salt. Use systemd-rpm-macros and user unit preset file.

## Status

Ready for review

## Description of Changes
Towards #1052
Fixes #1006 

- Use systemd-rpm-macros in rpm .spec and enable user units according to a preset file.  
- Manage sdw-notify.timer (and all systemd user) units via systemd macros in rpm %post (and %preun) instead of via Salt, and stop using `--global` for enabling user units.

## Testing

- [x] Visual review
- [ ] Manual testing (see below)
- Start from an environment without a dom0 workstation config and without any securedrop systemd services configured in dom0. (This could be an `--uninstall`, or it could be sneakily removing the dom0 rpm and disabling then removing (deleting) the sd systemd services `securedrop-user-xfce-icon-size`, `securedrop-user-xfce-settings`, and `sdw-notify.timer` and `sdw-notify.service`. Reload the systemd user daemon (systemctl --user daemon-reload) and ensure those services are not listed.
- Build an rpm from the tip of this branch (`make build-rpm`), and copy it into dom0, then install it. Reboot the laptop.  
- [ ]  `systemctl --user status` for each of `sdw-notify.timer`, `securedrop-user-xfce-icon-size`, and `securedrop-user-xfce-settings` shows them as enabled, and `systemctl status securedrop-logind-override-disable` enabled (on dev systems the service will run; on staging and prod systems the expected behaviour is that the service will fail a ConditionPath check and be disabled.)
- [ ] When the rpm is uninstalled, securedrop-related systemd units (user and system) are all removed.
- Upgrade test plan: starting from a preinstalled securedrop-workstation-dom0-config and an existing workstation setup, confirm that desired services are loaded, as above.
- Build rpm from the tip of this branch and install over top of your existing setup.
- [x] RPM installation completes successfully
- [x] Desired services are still in place, as above. 

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing pilot instances
2. New installs

## Checklist

### If you have made changes to the provisioning logic


### If you have added or removed files


### If documentation is required
